answerking_app/services/order_service.py

```diff
@@ -86,57 +77,48 @@ def create(body: OrderCreateDict) -> Order | None:
 
 
 def update(order: Order, body: OrderUpdateDict) -> Order | None:
-    try:
-        address: str = validate_address_string(body["address"])
-        order.address = address
-    except ValidationError:
-        return None
-    except KeyError:
-        pass
-
-    try:
-        updated_status: str = body["status"]
-        status: Status = Status.objects.get(status=updated_status)
-        order.status = status
-    except Status.DoesNotExist:
+    serialized_order = ClientOrderSerializer(data=body)
+    if not serialized_order.is_valid():
         return None
-    except KeyError:
-        pass
-
+    order.address = serialized_order.data["address"]
     order.save()
 
     return order
 
 
-def add_item(order: Order, item: Item, quantity: int) -> Order | None:
-    try:
-        quantity = validate_positive_number(quantity)
-        if quantity == 0:
-            return remove_item(order, item)
-
-        sub_total: Decimal = validate_price(quantity * item.price)
-    except ValidationError:
+def add_item(
+    order: Order, item: Item, body: QuantityUpdateDict | StatusUpdateDict
+) -> Order | None:
+    serialized_body = ClientOrderInfoUpdateSerializer(data=body)
+    if not serialized_body.is_valid():
         return None
+    quantity: int | None = serialized_body.data.get("quantity", None)
+    status: str | None = serialized_body.data.get("status", None)
 
-    if item not in order.order_items.all():
+    if quantity == 0:
+        remove_item(order, item)
+    elif quantity and item not in order.order_items.all():
         try:
             order.order_items.add(
                 item,
                 through_defaults={
                     "quantity": quantity,
-                    "sub_total": sub_total,
+                    "sub_total": item.price * quantity,
                 },
             )
         except DataError:
             return None
-    else:
+    elif quantity:
         try:
             order.orderline_set.filter(item=item.id).update(
-                quantity=quantity, sub_total=sub_total
+                quantity=quantity, sub_total=quantity * item.price
             )
         except DataError:
             return None
 
```

Comment:
should there be a statement in the if block handling if quantity is None?

Reply:
no, this was a bug, I mixed 2 things, the quantity will never be None since it's handled by the serializer, I fixed this with the last push